test/common/user_permissions.go

```diff
@@ -0,0 +1,227 @@
+		}
+		if p.Name != expectedRhmiDeveloperNamespace {
+			t.Fatal(fmt.Sprintf("test failed - found project for rhmi-developer does not match expected : %s", expectedRhmiDeveloperNamespace))
+		}
+	}
+	t.Log("test-passed - found projects for rhmi-developer are as expected")
+
+	// get dedicated admin token
+	dedicatedAdminToken, err := resources.GetUserOpenshiftToken(oauthRoute.Spec.Host, masterURL, resources.DefaultIDP, "customer-admin01", "Password1")
+	if err != nil {
+		t.Fatalf("error occured trying to get token : %v", err)
+	}
+
+	// get projects for dedicated admin
+	dedicatedAdminFoundProjects, err := ocGetProjects(masterURL, dedicatedAdminToken)
+	if err != nil {
+		t.Fatalf("error occured while getting user projects : %v", err)
+	}
+
+	// check if projects are as expected for rhmi developer
+	if result := verifyFoundProjects(dedicatedAdminFoundProjects.Items); !result {
+		t.Fatal("test-failed - projects missing for dedicated-admins")
+	}
+	t.Log("test-passed - found projects for dedicated-admins are as expected")
+
+	// get fuse pods for rhmi developer
+	podlist, err := ocGetNameSpacePods(masterURL, pathFusePods, rhmiDevUserToken)
+	if err != nil {
+		t.Fatalf("error occured while getting pods : %v", err)
+	}
+
+	// check if six pods are running
+	runningCount := 0
+	for _, p := range podlist.Items {
+		if p.Status.Phase == "Running" {
+			runningCount++
+		}
+	}
+	if runningCount != expectedFusePodCount {
+		t.Fatalf("test-failed - %d found fuse pod count not match expected count of %d", runningCount, expectedFusePodCount)
+	}
+	t.Logf("test-passed - found expected %d running pods in fuse namespace", expectedFusePodCount)
+
+	// log through rhmi developer fuse podlist
+	for _, p := range podlist.Items {
+		if p.Status.Phase == "Running" {
+			logOpt := LogOptions{p.Spec.Containers[0].Name, "false", "10"}
+			lv, err := query.Values(logOpt)
+			if err != nil {
+				t.Fatal(err)
+			}
+			// verify an rhmi developer can access the pods logs
+			if err := verifySuccessfullHTTPRequest(fmt.Sprintf("%s/%s/%s/log?%s", masterURL, pathFusePods, p.Name, lv.Encode()), rhmiDevUserToken); err != nil {
+				t.Fatalf("test-failed - rhmi devolper unable to access fuse logs at %s, error : %v", p.Name, err)
+			}
+		}
+	}
+}
+
+// verifies that there is at least 1 project with a prefix `openshift` , `redhat` and `kube`
+func verifyFoundProjects(projects []projectv1.Project) bool {
+	var hasOpenshiftPrefix, hasRedhatPrefix, hasKubePrefix bool
+	for _, ns := range projects {
+		if strings.HasPrefix(ns.Name, "openshift") {
+			hasOpenshiftPrefix = true
+		}
+		if strings.HasPrefix(ns.Name, "redhat") {
+			hasRedhatPrefix = true
+		}
+		if strings.HasPrefix(ns.Name, "kube") {
+			hasKubePrefix = true
+		}
+	}
+	return hasKubePrefix && hasRedhatPrefix && hasOpenshiftPrefix
+}
+
+// makes call to url, and checks if the resp is successful, returns error if request is not successful
+func verifySuccessfullHTTPRequest(url, token string) error {
```

Comment:
Typo in the function name here `Successful`. Again I think we may need to make this function name more descriptive. As it's for an OpenShift request (or at least sets a cookie for OpenShift based on the provided token) and only performs a GET. e.g.

```
func doOpenshiftGetRequest(url, token string) error {
...
}
```

We could probably also consider using `ocRequest` inside this function, right? Or maybe removing this altogether and doing the 200 check inline as it's only used in one place currently [1]?

[1] https://github.com/integr8ly/integreatly-operator/pull/510/files#diff-036b3912c851d760c7f05430eee9e3d8R127